devops/girder/annotation_client/annotation_client/annotations.py

```diff
@@ -46,8 +46,9 @@ def getAnnotationsByDatasetId(self, datasetId, shape=None, limit=50, offset=0):
         """
         url = PATHS['annotation_by_dataset'].format(datasetId=datasetId)
 
+        url = f'{url}&limit={limit}&offset{offset}'
         if shape:
-            url = '{url}&shape={shape}&limit={limit}&offset{offset}'.format(url=url, shape=shape, limit=limit, offset=offset)
+            url = f'{url}&shape={shape}'
```

Comment:
I think you should add a else case here

Reply:
What should I put in the else case?

Reply:
I think that if shape is not given, it should just return all annotations (subject to limit and offset).
Also, while we are at it, is there a way to add a filter by tag? Like, to get all annotations by tag? That would be useful.
Also, let's change default limit to 1,000,000. I think that is a reasonable number. 50 is pretty small.

Reply:
Returning all annotations and taking limit and offset into account is now the default behavior when no shape is given
I can add the feature to filter by tags and increase the default limit
In the tag filtering, what would be the algorithm?
- Get all annotations with exactly the given tags
- Get all annotations with all the given tags and eventually other tags
- Get all annotations with at least one of the given tags

Reply:
Yeah, I wasn't sure exactly what to do there :). I think it should be all of the given tags and potentially other additional tags.